site/snippets/session-keys/supported-permissions.ts

```diff
@@ -0,0 +1,175 @@
+  limit: nativeTokenSpendLimit, // The limit's maximum value. If a refresh interval is set, this is the max per interval.
+  limitUsed: nativeTokenSpendLimitUsed, // How much of the limit is used. If a refresh interval is set, this is the amount used in the current interval.
+  refreshInterval: nativeTokenRefreshInterval, // How often to reset the limit and start counting again. If zero, never refresh the limit.
+  lastUsedTime: nativeTokenLastUsedTime, // The start of the latest interval, if using the refresh interval.
+} = nativeTokenSpendingLimit;
+
+// The spending limit for an ERC-20 token.
+const erc20SpendingLimit = await sessionKeyPluginView.getERC20SpendLimitInfo([
+  accountAddress,
+  sessionKeyAddress,
+  exampleERC20Address,
+]);
+
+const {
+  hasLimit: hasErc20TokenSpendLimit, // Whether or not an ERC-20 token spending limit is enforced on the session key for this token address.
+  limit: erc20TokenSpendLimit, // The limit's maximum value. If a refresh interval is set, this is the max per interval.
+  limitUsed: erc20TokenSpendLimitUsed, // How much of the limit is used. If a refresh interval is set, this is the amount used in the current interval.
+  refreshInterval: erc20TokenRefreshInterval, // How often to reset the limit and start counting again. If zero, never refresh the limit.
+  lastUsedTime: erc20TokenLastUsedTime, // The start of the latest interval, if using the refresh interval.
+} = erc20SpendingLimit;
+
+// - The spending limit on gas for a given session key, measured in wei.
+// - Whether or not the spending limit will reset in the next interval, if a refresh interval is set.
+const [gasSpendingLimit, shouldReset] =
+  await sessionKeyPluginView.getGasSpendLimit([
+    accountAddress,
+    sessionKeyAddress,
+  ]);
+
+const {
+  hasLimit: hasGasSpendLimit, // Whether or not a gas spending limit is enforced on the session key
+  limit: gasSpendLimit, // The limit's maximum value. If a refresh interval is set, this is the max per interval.
+  limitUsed: gasSpendLimitUsed, // How much of the limit is used. If a refresh interval is set, this is the amount used in the current interval.
+  refreshInterval: gasRefreshInterval, // How often to reset the limit and start counting again. If zero, never refresh the limit.
+  lastUsedTime: gasLastUsedTime, // The start of the latest interval, if using the refresh interval.
+} = gasSpendingLimit;
+
+// The paymaster address required for a given session key.
+// If there is no required paymaster, this will return the zero address.
+const requiredPaymaster = await sessionKeyPluginView.getRequiredPaymaster([
+  accountAddress,
+  sessionKeyAddress,
+]);
+// #endregion view-permissions
+
+// To suppress typescript errors about unused variables from destructuring, "use" the variables here.
+// For some reason, this is an error, and not just a warning. (TS error 6198)
+
+// eslint-disable-next-line @typescript-eslint/no-unused-expressions
```

Comment:
One possibility to suppress these warnings that might feel more "natural" for docs readers would be to log these values throughout, e.g. after the line that loads `hasNativeTokenSpendLimit` do
```ts
console.log("hasNativeTokenSpendLimit:", hasNativeTokenSpendLimit);
```